src/Microsoft.Azure.Batch.SoftwareEntitlement.Server/Controllers/SoftwareEntitlementsController.cs

```diff
@@ -32,11 +31,18 @@ public class SoftwareEntitlementsController : Controller
         /// <param name="serverOptions">Options to use when handling requests.</param>
         /// <param name="logger">Reference to our logger for diagnostics.</param>
         /// <param name="lifetime">Lifetime instance to allow us to automatically shut down if requested.</param>
-        public SoftwareEntitlementsController(ServerOptions serverOptions, ILogger logger, IApplicationLifetime lifetime)
+        /// <param name="verifier">A component responsible for checking data in the request against the claims
+        /// in the token</param>
+        public SoftwareEntitlementsController(
+            ServerOptions serverOptions,
+            ILogger logger,
+            IApplicationLifetime lifetime,
```

Comment:
@Porges suggested (much earlier) that instead of injecting `IApplicatoinLifetime` here, it could be set up in the overall pipeline for the HTTP Server itself. I haven't investigated how to do that, but if you want a challenge @peterbom ... 😉